lib/membrane/rtp/tcp_depayloader.ex

```diff
@@ -0,0 +1,119 @@
+defmodule Membrane.RTP.TCP.Depayloader do
+  @moduledoc """
+  This element provides functionality of depayloading RTP Packets received by TCP and redirecting
+  RTSP messages received in the same stream. The encapsulation is described in RFC 7826 Section 14.
+  """
+  use Membrane.Filter
+
+  alias Membrane.{Buffer, RemoteStream, RTP, RTSP}
+
+  def_options rtp_channel_id: [
+                spec: non_neg_integer(),
+                default: 0,
+                description: """
+                Channel identifier which encapsulated RTP packets will have.
+                """
+              ],
+              rtsp_session: [
+                spec: pid() | nil,
+                default: nil,
+                description: """
+                PID of a RTSP Session (returned from Membrane.RTSP.start or Membrane.RTSP.start_link)
+                that received RTSP responses will be forwarded to. If nil the responses will be
+                discarded.
+                """
+              ]
+
+  def_input_pad :input, accepted_format: %RemoteStream{type: :bytestream}
+
+  def_output_pad :output, accepted_format: %RemoteStream{type: :packetized, content_format: RTP}
+
+  @impl true
+  def handle_init(_ctx, opts) do
+    state =
+      Map.from_struct(opts)
+      |> Map.merge(%{
+        unprocessed_data: <<>>
+      })
+
+    {[], state}
+  end
+
+  @impl true
+  def handle_stream_format(:input, _stream_format, _ctx, state) do
+    stream_format = %RemoteStream{type: :packetized, content_format: RTP}
+    {[stream_format: {:output, stream_format}], state}
+  end
+
+  @impl true
+  def handle_buffer(
+        :input,
+        %Buffer{payload: payload, metadata: metadata},
+        _ctx,
+        %{unprocessed_data: unprocessed_data, rtsp_session: rtsp_session} = state
+      ) do
+    unprocessed_data =
+      if rtsp_response?(unprocessed_data, payload) do
+        if rtsp_session != nil do
+          {:ok, %RTSP.Response{status: 200}} =
+            RTSP.handle_response(rtsp_session, unprocessed_data)
+        end
+
+        <<>>
+      else
+        unprocessed_data
+      end
+
+    packets_binary = unprocessed_data <> payload
+
+    {unprocessed_data, complete_packets_binaries} =
+      get_complete_packets(packets_binary, state.rtp_channel_id)
+
+    packets_buffers =
+      Enum.map(complete_packets_binaries, &%Buffer{payload: &1, metadata: metadata})
+
+    {[buffer: {:output, packets_buffers}], %{state | unprocessed_data: unprocessed_data}}
+  end
+
+  @spec rtsp_response?(binary(), binary()) :: boolean()
+  defp rtsp_response?(maybe_rtsp_response, new_payload) do
+    # If new payload starts with a "$" and unprocessed data started with "RTSP" then we know
+    # the unprocessed data should be a RTSP message
+    String.starts_with?(new_payload, "$") and String.starts_with?(maybe_rtsp_response, "RTSP")
```

Comment:
I think this piece of code is self explanatory - I would rather have a brief explanation what `$` means :D